src/main/java/com/google/growpod/data/User.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.growpod.data;
+
+/** User data class. */
+public class User {
+
+  public static final int MAX_BIOGRAPHY_LEN = 160;
+
+  /** A unique id. */
+  private String id;
+
+  private String email;
+  /** This is the only name we track. */
+  private String preferredName;
+
+  private String biography;
+  /**
+   * We do not wish to keep exact user locations -- this will store some approximate location data.
+   * Unfortunately, this will only work in the US.
+   */
+  private String zipCode;
+
+  /**
+   * Constructs a new User.
+   *
+   * @param id A unique ID for the user. This value must be supplied by the user.
+   * @param email The user's email.
+   * @param preferredName The user's preferred name.
+   * @param biography A user-submitted biography truncated to MAX_BIOGRAPHY_LEN.
+   * @param zipCode The user's postal code.
+   */
+  public User(String id, String email, String preferredName, String biography, String zipCode) {
+    this.id = id;
+    this.email = email;
+    this.preferredName = preferredName;
+    // Truncate biography
+    this.biography = biography.substring(0, Math.min(biography.length(), MAX_BIOGRAPHY_LEN));
```

Comment:
Consider throwing an error here -- otherwise a user may not understand why their bio was silently truncated.

Reply:
I think I will temporarily remove this validation to make it consistent with how other fields are *not* validated, but I think  will add more comprehensive validation with error handling once I write a `POST` method that actually creates a user.

Reply:
I will absolutely do something like this when I reimplement validation later on